databricks/koalas/indexes.py

```diff
@@ -399,6 +400,138 @@ def copy(self, name=None):
             result.name = name
         return result
 
+    def value_counts(
+        self, normalize=False, sort=True, ascending=False, dropna=True
+    ):
+        """
+        Return a Series containing counts of unique values.
+        The resulting object will be in descending order so that the
+        first element is the most frequently-occurring element.
+        Excludes NA values by default.
+
+        Parameters
+        ----------
+        normalize : boolean, default False
+            If True then the object returned will contain the relative
+            frequencies of the unique values.
+        sort : boolean, default True
+            Sort by frequencies.
+        ascending : boolean, default False
+            Sort in ascending order.
+        dropna : boolean, default True
+            Don't include counts of NaN.
+
+        Returns
+        -------
+        Series
+
+        See Also
+        --------
+        Series.count: Number of non-NA elements in a Series.
+        DataFrame.count: Number of non-NA elements in a DataFrame.
+
+        Examples
+        --------
+
+        >>> s = ks.Series([0, 1, 2, 3, 4, 5], index=[3, 1, 2, 3, 4, np.nan])
+        >>> s.index
+        Float64Index([3.0, 1.0, 2.0, 3.0, 4.0, nan], dtype='float64')
+
+        >>> s.index.value_counts().sort_index()
+        1.0    1
+        2.0    1
+        3.0    2
+        4.0    1
+        Name: count, dtype: int64
+
+        **sort**
+
+        With `sort` set to `False`, the result wouldn't be sorted by number of count.
+
+        >>> s.index.value_counts(sort=True).sort_index()
+        1.0    1
+        2.0    1
+        3.0    2
+        4.0    1
+        Name: count, dtype: int64
+
+        **normalize**
+
+        With `normalize` set to `True`, returns the relative frequency by
+        dividing all values by the sum of values.
+
+        >>> s.index.value_counts(normalize=True).sort_index()
+        1.0    0.2
+        2.0    0.2
+        3.0    0.4
+        4.0    0.2
+        Name: count, dtype: float64
+
+        **dropna**
+
+        With `dropna` set to `False` we can also see NaN index values.
+
+        >>> s.index.value_counts(dropna=False).sort_index()
```

Comment:
Seems this test is failed:

```
____________ [doctest] databricks.koalas.indexes.Index.value_counts ____________
530         2.0    0.2
531         3.0    0.4
532         4.0    0.2
533         Name: count, dtype: float64
534 
535         **dropna**
536 
537         With `dropna` set to `False` we can also see NaN index values.
538 
539         >>> s.index.value_counts(dropna=False).sort_index()
Differences (unified diff with -expected +actual):
    @@ -1,6 +1,6 @@
    -1.0    1
    -2.0    1
    -3.0    2
    -4.0    1
    -NaN    1
    + 1.0    1
    + 2.0    1
    + 3.0    2
    + 4.0    1
    +NaN     1
     Name: count, dtype: int64
```

You can skip this test for now I think.

Reply:
ah, now i remember why i skipped those tests.
Thanks for the comment! 😄